src/core/streams/stream_actions.js

```diff
@@ -36,5 +38,87 @@ export const StreamActions = {
 
   refresh() {
     session.refresh(this.baseURI, this.requestId)
+  },
+
+  morph() {
+    this.targetElements.forEach((targetElement) => {
+      try {
+        const morphStyle = this.getAttribute("data-turbo-morph-style") || "outerHTML"
+        morph(targetElement, this.templateContent, {
+          morphStyle: morphStyle,
+          ignoreActiveValue: true,
+          callbacks: {
+            beforeNodeAdded,
+            beforeNodeMorphed,
+            beforeAttributeUpdated,
+            beforeNodeRemoved,
+            afterNodeMorphed,
+          },
+        })
+        dispatchEvent("turbo:morph", {
+          currentElement: targetElement,
+          newElement: this.templateContent,
+        })
+      } catch (error) {
+        console.error(error)
+      }
+    })
+  },
+}
+
+const dispatchEvent = (eventName, detail, cancelable = false) => {
```

Comment:
`dispatchEvent` is a js API https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent

Also, you might want to follow their conventions and use `dispatch` directly ☺️

Reply:
Hey @marcelokanzaki, I am aware of that, but this is scoped locally, so it won't collide with the global `dispatchEvent`. The file is small and clean, and I'm trying to keep it as tight as possible. I didn't want to repeat the same object all over the place; it just doesn't look as good 😄 however if there is a strong objection against this pattern Im open to changing it.

Reply:
@marcelokanzaki I took some time to consider your feedback, ended up refactoring to your suggestion 👍🏻